### PR TITLE
Add basic client and auth code

### PR DESCRIPTION
**Pull request by stuart-mclaren-hpe**
These clients will be used to access various APIs. 

For example the virt client will be used for creating Virtual Machines.

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	"context"
+	"time"
+
+	"github.com/HewlettPackard/hpegl-pcbe-terraform-resources/internal/auth"
+	"github.com/HewlettPackard/hpegl-pcbe-terraform-resources/internal/defaults"
+	"github.com/HewlettPackard/hpegl-pcbe-terraform-resources/internal/sdk/async"
+	"github.com/HewlettPackard/hpegl-pcbe-terraform-resources/internal/sdk/systems"
+	"github.com/HewlettPackard/hpegl-pcbe-terraform-resources/internal/sdk/virt"
+	"github.com/microsoft/kiota-abstractions-go/authentication"
+	"github.com/microsoft/kiota-http-go"
```
Comment (AislingHPE) [inline internal/client/client.go:13]: While looking at the very very long line i noticed, does this need to be `nethttplibrary "whatever"` to compile?

Comment (stuart-mclaren-hpe) [reply internal/client/client.go:13]: The ci build job is passing.

Comment (AislingHPE) [reply internal/client/client.go:13]: spooky, wonder how that works

Comment (AislingHPE) [reply internal/client/client.go:13]: oh that's the name of it's 'main' package, TIL

**Review by AislingHPE — DISMISSED** [review]

**Review by AislingHPE — APPROVED** [review]

**Pull request merged**